src/internal/connector/graph/middleware.go

```diff
@@ -232,15 +270,22 @@ func (middleware RetryHandler) retryRequest(
 			err)
 	}
 
-	if respErr != nil {
-		return nil, Stack(ctx, respErr).With("retry_count", executionCount)
+	if initialErr != nil {
+		return nil, stackReq(ctx, req, nil, initialErr)
 	}
 
 	return resp, nil
 }
 
-func (middleware RetryHandler) isRetriableErrorCode(req *http.Request, code int) bool {
-	return code == http.StatusInternalServerError || code == http.StatusServiceUnavailable
+var retryableRespCodes = []int{
+	http.StatusInternalServerError,
+	http.StatusServiceUnavailable,
```

Comment:
Reading MSFT's guidance on how to handle service unavailable errors. 
`This is likely because the services is busy. You should employ a back-off strategy similar to 429. Additionally, you should always make new retry requests over a new HTTP connection.`

Just to confirm, will retries be done over a new HTTP connection?

Reply:
^ Taken from https://learn.microsoft.com/en-us/graph/best-practices-concept#handling-expected-errors?WT.mc_id=M365-MVP-5002900

Reply:
I think so?  iirc, default behavior for the graph client is to construct new connections on each call (barring more extreme circumstances that easily allow re-use, such as long dormancy with all prior connections being closed).